### DISCUSSION TITLE
Should PayloadCMS be NPM first?

### DISCUSSION THREAD

**Discussion by TomSmedley** [payloadcms/payload#1182] [General]
Hi everyone,

So I've personally avoided tools like yarn and pnpm, I know these are much faster than npm but how fast my packages install in the grand scheme of a project isn't going to make me miss any deadlines.

What I have found is PayloadCMS goes on the premise that everyone is using yarn. But we absolutely know that if someone is using node they have npm, assuming they also have yarn I think works against the DX.

For example `npm run build` does `yarn copyfiles...` I personally think this should be swapped to `npm run copyfiles...` by default. Also another thing when I'm deploying through Docker I also have to install yarn.

What do you think?

**Comment by qivox** [comment]
First of all, package managers like yarn or pnpm do not only exist because they install packages faster. They do way more than that but that's not the point here.

You can simply swap out the payload scripts in your package.json with your preferred package manager exactly they way you suggested. You are not forced to use yarn at all. Take a look at their [Remix](https://github.com/payloadcms/remix-server) repo which uses pnpm for example.

**Comment by denolfe** [comment]
Thank you for the thoughts @TomSmedley. Dev experience is one of the main focuses of Payload, so we make an ongoing effort to ensure that Payload runs well with any package manager.

Other than your Docker example, are there any other issues that you run into with npm? Like @qivox said, you should be able to modify these scripts for your own needs.

**Reply by TomSmedley** [reply]
It's just an idea that it ships with sensible defaults as in, npm will always be available and if people wanted to use yarn they could edit the package.json. I'm just preempting as PayloadCMS gains more attraction from developers of all skill levels (which it will) having trouble building out their application.

To me it seems more logical to ship with the defaults which we know will be in place rather than making assumptions.
